### ISSUE TITLE
New tests without baseline images never produce a result

### ISSUE THREAD

**Issue by QuLogic** [matplotlib/matplotlib#7240]
When adding a new test, you don't have a baseline image, and probably expect to copy this out of the result set. On master, the result is never created if the baseline image doesn't exist. It appears that [`copy_baseline` will raise an expected failure](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/testing/decorators.py#L282) before the result is saved.

Relatedly, pytest does not have an easy way of seeing this message that the baseline image is missing. Using `pytest -v`, it just marks the test as `xfail` instead of `x`. With the old `tests.py -v`, it will print out a much more useful `KNOWNFAIL: Do not have baseline image .../result_images/test_mplot3d/proj3d_lines_dists-expected.png because this file does not exist: .../mpl_toolkits/tests/baseline_images/test_mplot3d/proj3d_lines_dists.png`.

From what I've seen, a missing baseline image has been marked as xfail since before pytest, but I'm not sure that makes sense; I think it should be an error.

I think this is related to some pytest re-factorings; cc @Kojoley.

**Comment by Kojoley** [comment]
```
>python tests.py -v matplotlib.tests.test_axes
Python byte-compilation optimization level: 0
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-png] ... KNOWNFAIL: Do not have baseline image ...\result_images\test_axes\formatter_ticker_001-expected.png because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_001.png
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-pdf] ... KNOWNFAIL: Do not have baseline image ...\result_images\test_axes\formatter_ticker_001-expected.pdf because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_001.pdf
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-svg] ... KNOWNFAIL: Do not have baseline image ...\result_images\test_axes\formatter_ticker_001-expected.svg because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_001.svg
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[1-formatter_ticker_002-png] ... KNOWNFAIL: Do not have baseline image ...\result_images\test_axes\formatter_ticker_002-expected.png because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_002.png
...
```

```
>py.test -vra lib/matplotlib/tests/test_axes.py
============================= test session starts =============================
platform win32 -- Python 3.5.2, pytest-3.0.1, py-1.4.31, pluggy-0.3.1 -- C:\Miniconda2\envs\test-mpl-py3\python.exe
cachedir: .cache
rootdir: ..., inifile: pytest.ini
plugins: cov-2.3.1, mpl-0.5, timeout-1.0.0, xdist-1.14
collected 436 items

lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-png] <- ...\lib\matplotlib\tests\test_axes.py xfail
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-pdf] <- ...\lib\matplotlib\tests\test_axes.py xfail
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-svg] <- ...\lib\matplotlib\tests\test_axes.py xfail
lib/matplotlib/tests/test_axes.py::test_formatter_ticker[1-formatter_ticker_002-png] <- ...\lib\matplotlib\tests\test_axes.py xfail
=========================== short test summary info ===========================
XFAIL lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-png]
  reason: Do not have baseline image ...\result_images\test_axes\formatter_ticker_001-expected.png because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_001.png
XFAIL lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-pdf]
  reason: Do not have baseline image ...\result_images\test_axes\formatter_ticker_001-expected.pdf because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_001.pdf
XFAIL lib/matplotlib/tests/test_axes.py::test_formatter_ticker[0-formatter_ticker_001-svg]
  reason: Do not have baseline image ...\result_images\test_axes\formatter_ticker_001-expected.svg because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_001.svg
XFAIL lib/matplotlib/tests/test_axes.py::test_formatter_ticker[1-formatter_ticker_002-png]
  reason: Do not have baseline image ...\result_images\test_axes\formatter_ticker_002-expected.png because this file does not exist: ...\lib\matplotlib\tests\baseline_images\test_axes\formatter_ticker_002.png
```

**Comment by QuLogic** [comment]
Ah, so `-ra` is needed to get the full reason. Nevertheless, the result images are not saved, which is the main point here.

**Comment by Kojoley** [comment]
> Ah, so `-ra` is needed to get the full reason. 

Yes.
```
  -r chars              show extra test summary info as specified by chars
                        (f)ailed, (E)error, (s)skipped, (x)failed, (X)passed,
                        (p)passed, (P)passed with output, (a)all except pP.
```

> Nevertheless, the result images are not saved, which is the main point here.

It should be fixed with #7245. Could you test it just to be sure?
